osquery/tables/system/linux/services.cpp

```diff
@@ -0,0 +1,585 @@
+
+  return enabled;
+}
+
+bool isUpstartEnabled() {
+  auto enabled = boostfs::exists("/etc/init");
+  return enabled;
+}
+
+bool isSystemdEnabled() {
+  static auto enabled = loadSystemdDependencies();
+  VLOG(1) << "systemd support has been " << (enabled ? "enabled" : "disabled");
+
+  return enabled;
+}
+
+Status enumerateSysVinitServices(QueryData& query_data) {
+  static bool sysvinit_enabled = isSysVinitEnabled();
+  if (!sysvinit_enabled) {
+    return Status(0, "OK");
+  }
+
+  static const std::vector<std::string> run_level_dir_list = {"/etc/rc0.d",
+                                                              "/etc/rc1.d",
+                                                              "/etc/rc2.d",
+                                                              "/etc/rc3.d",
+                                                              "/etc/rc4.d",
+                                                              "/etc/rc5.d",
+                                                              "/etc/rc6.d",
+                                                              "/etc/rcS.d"};
+
+  std::unordered_map<std::string, std::vector<size_t>> service_list;
+
+  for (size_t run_level = 0U; run_level < run_level_dir_list.size();
+       ++run_level) {
+    const auto& run_level_path = run_level_dir_list[run_level];
+
+    try {
+      boostfs::directory_iterator end_it;
+      for (auto it = boostfs::directory_iterator(run_level_path); it != end_it;
+           ++it) {
+        const auto& symlink = it->path();
+        if (!boostfs::is_symlink(symlink)) {
+          continue;
+        }
+
+        auto symlink_destination = boostfs::canonical(symlink);
+        if (!boostfs::is_regular_file(symlink_destination)) {
+          VLOG(1) << "Skipping broken symlink: " << symlink << " -> "
+                  << symlink_destination;
+
+          continue;
+        }
+
+        auto service_name = symlink_destination.filename().string();
+        service_list[service_name].push_back(run_level);
+      }
+    } catch (const boostfs::filesystem_error& e) {
+      VLOG(1) << "An error has occurred: " << e.what() << ". Continuing anyway";
+    }
+  }
+
+  for (const auto& p : service_list) {
+    const auto& service_name = p.first;
+    const auto& run_level_list = p.second;
+
+    Row r;
+    r["name"] = service_name;
+    r["status"] = "";
+    r["path"] = std::string("/etc/init.d/") + service_name;
+    r["service_type"] = "SysVinit";
+
+    // Use the same syntax as Upstart
+    if (!run_level_list.empty()) {
+      r["start_type"] = "runlevel [";
+
+      for (const auto& run_level : run_level_list) {
+        if (run_level != 7) {
+          r["start_type"] += std::to_string(run_level);
+        } else {
+          r["start_type"] += "S";
+        }
+      }
+
+      r["start_type"] += "]";
+    }
+
+    query_data.push_back(r);
+  }
+
+  return Status(0, "OK");
+}
+
+sd_bus* getSystemdBusHandle() {
+  /*
```

Comment:
Aesthetically the code base uses `/**` to start long-form comments and `  * ` to continue on new-lines, similar to long-form docblocks.